gcloud/resource_manager/project.py

```diff
@@ -0,0 +1,323 @@
+
+    .. note::
+
+        It's unlikely that you'd need to instantiate this outside the context
+        of a :class:`.client.Client`, so in general, it's best to get a Project
+        from a Resource Manager client.
+
+    To create a new project::
+
+        >>> from gcloud import resource_manager
+        >>> client = resource_manager.Client()
+        >>> project = client.project('purple-spaceship-123')
+        >>> project.name = 'Purple Spaceship Project!'
+        >>> project.create()
+
+    To get an existing project::
+
+        >>> from gcloud import resource_manager
+        >>> client = resource_manager.Client()
+        >>> project = client.get_project('purple-spaceship-123')
+        >>> print project.name
+        Purple Spaceship Project!
+
+    To manage labels::
+
+        >>> from gcloud import resource_manager
+        >>> client = resource_manager.Client()
+        >>> project = client.get_project('purple-spaceship-123')
+        >>> project.labels = {'color': 'purple'}
+        >>> project.labels['environment'] = 'production'
+        >>> project.update()
+
+    See:
+    https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects
+
+    :type project_id: string
+    :param project_id: the globally unique id of the project
+    """
+    def __init__(self, client, project_id):
+        self._client = client
+        self.project_id = project_id
+        self.name = None
+        self.number = None
+        self.labels = {}
+        self.status = None
+
+    def __repr__(self):
+        if self.name:
+            display_name = '%s (%s)' % (self.name, self.project_id)
+        else:
+            display_name = self.project_id
+        return '<Project: %s>' % display_name
+
+    @classmethod
+    def from_api_repr(cls, resource, client):
+        """Factory:  construct a project given its API representation.
+
+        :type resource: dict
+        :param resource: project resource representation returned from the API
+
+        :rtype: :class:`gcloud.resource_manager.project.Project`
+        """
+        project = cls(project_id=resource['projectId'], client=client)
+        project.set_properties_from_api_repr(resource)
+        return project
+
+    def set_properties_from_api_repr(self, resource):
+        """Update specific properties from its API representation."""
+        self.name = resource.get('name')
+        self.number = resource['projectNumber']
+        self.labels = resource.get('labels', {})
+        self.status = resource['lifecycleState']
+
+    @property
+    def full_name(self):
+        """Fully-qualified name (ie, ``'projects/purple-spaceship-123'``)."""
+        if not self.project_id:
+            raise ValueError('Missing project ID!')
+        return 'projects/%s' % (self.project_id)
+
+    @property
+    def path(self):
+        """URL for the project (ie, ``'/projects/purple-spaceship-123'``)."""
+        return '/%s' % (self.full_name)
+
+    @property
+    def client(self):
```

Comment:
The only reason to keep it as a property would be to prevent people from setting it, but still address it as `.client` rather than `._client`. I could go either way -- I thought we wanted to discourage people from swapping clients .. if we don't care about that, we can just keep it as `.client`.

Let me know what you think is best.